applications/IgaApplication/custom_utilities/create_breps_sbm_utilities.h

```diff
@@ -0,0 +1,492 @@
+                    active_range_vector[1] = surrogate_coords_y_inner[(i + 1) % surrogate_coords_x_inner.size()];
+                }
+                else {
+                    active_range_vector[0] = surrogate_coords_x_inner[i];
+                    active_range_vector[1] = surrogate_coords_x_inner[(i + 1) % surrogate_coords_x_inner.size()];
+                }
+                bool curve_direction = true;
+
+                // re-order
+                if (active_range_vector[0] > active_range_vector[1]) {
+                    double temp = active_range_vector[1];
+                    active_range_vector[1] = active_range_vector[0] ;
+                    active_range_vector[0] = temp ;
+                }
+
+                NurbsInterval brep_active_range(active_range_vector[0], active_range_vector[1]);
+
+                auto p_brep_curve_on_surface = Kratos::make_shared<BrepCurveOnSurfaceType>(
+                    pSurface, trimming_curves_inner[i], brep_active_range, curve_direction);
+                p_brep_curve_on_surface->SetId(id_brep_curve_on_surface);
+                rModelPart.AddGeometry(p_brep_curve_on_surface);
+                id_brep_curve_on_surface++;
+                trimming_brep_curve_vector[i] = p_brep_curve_on_surface ;
+
+            }
+
+        }
+    } 
+    
+    /**
+     * @brief Create a Single Brep object
+     * 
+     * @param firstBrepPoint 
+     * @param secondBrepPoint 
+     * @param activeRangeKnotVector 
+     * @return NurbsCurveGeometry<2, PointerVector<Point>>::Pointer 
+     */
+    static typename NurbsCurveGeometry<2, PointerVector<Point>>::Pointer CreateSingleBrep(
+        const Point::Pointer firstBrepPoint,
+        const Point::Pointer secondBrepPoint, 
+        const Vector activeRangeKnotVector)
+        {
+            // Create the data for the trimming curves
+            PointerVector<Point> control_points;
+            control_points.push_back(firstBrepPoint);
+            control_points.push_back(secondBrepPoint);
+            const int polynomial_degree = 1;
+            Vector knot_vector = ZeroVector(4) ;
+            knot_vector[0] = activeRangeKnotVector[0] ;
+            knot_vector[1] = activeRangeKnotVector[0] ;
+            knot_vector[2] = activeRangeKnotVector[1] ;
+            knot_vector[3] = activeRangeKnotVector[1] ;
+            // Create the trimming curves
+            typename NurbsCurveGeometry<2, PointerVector<Point>>::Pointer p_trimming_curve(
+                new NurbsCurveGeometry<2, PointerVector<Point>>(
+                    control_points,
+                    polynomial_degree,
+                    knot_vector));   
+            return p_trimming_curve;
+        }
+    
+    /**
+     * @brief Create a Brep Curves On Rectangle object
+     * 
+     * @param rModelPart 
+     * @param pSurfaceGeometry 
+     * @param A_uvw 
+     * @param B_uvw 
+     * @param lastGeometryId 
+     */
+    static void CreateBrepCurvesOnRectangle(ModelPart& rModelPart, 

```

Comment:
I know the intention of calling it `Rectangle`, but I will keep naming it `CreateBrepCurvesOnSurface` to avoid confusion.

Reply:
Okay. I get your point.